ecs-cli/modules/cli/compose/entity/service/service_test.go

```diff
@@ -601,3 +601,311 @@ func TestServiceRun(t *testing.T) {
+	expectedInput.serviceName = serviceName
+	expectedInput.deploymentConfig = &ecs.DeploymentConfiguration{
+		MaximumPercent:        aws.Int64(int64(deploymentMaxPercent)),
+		MinimumHealthyPercent: aws.Int64(int64(deploymentMinPercent)),
+	}
+
+	// call tests
+	upServiceWithCurrentTaskDefTest(t, cliContext, &config.CLIParams{}, &utils.ECSParams{}, expectedInput, existingService)
+	upServiceWithNewTaskDefTest(t, cliContext, &config.CLIParams{}, &utils.ECSParams{}, expectedInput, existingService)
+}
+
+func TestUpdateExistingServiceWithNewHCGP(t *testing.T) {
+	// define test flag set
+	healthCheckGracePeriod := 200
+
+	flagSet := flag.NewFlagSet("ecs-cli-up", 0)
+	flagSet.String(flags.HealthCheckGracePeriodFlag, strconv.Itoa(healthCheckGracePeriod), "")
+	cliContext := cli.NewContext(nil, flagSet, nil)
+
+	// define existing service
+	serviceName := "test-service"
+	existingService := &ecs.Service{
+		TaskDefinition: aws.String("arn/test-task-def"),
+		Status:         aws.String("ACTIVE"),
+		DesiredCount:   aws.Int64(0),
+		ServiceName:    aws.String(serviceName),
+		LoadBalancers:  []*ecs.LoadBalancer{}, // LB required for HCGP, but not verified before calling client
+	}
+
+	// define expected client input given the above info
+	expectedInput := getDefaultUpdateInput()
+	expectedInput.serviceName = serviceName
+	expectedInput.healthCheckGracePeriod = aws.Int64(int64(healthCheckGracePeriod))
+
+	// call tests
+	upServiceWithCurrentTaskDefTest(t, cliContext, &config.CLIParams{}, &utils.ECSParams{}, expectedInput, existingService)
+	upServiceWithNewTaskDefTest(t, cliContext, &config.CLIParams{}, &utils.ECSParams{}, expectedInput, existingService)
+}
+
+func TestUpdateExistingServiceWithDesiredCountOverOne(t *testing.T) {
+	// define test values
+	existingDesiredCount := 2
+
+	flagSet := flag.NewFlagSet("ecs-cli-up", 0)
+	cliContext := cli.NewContext(nil, flagSet, nil)
+
+	// define existing service
+	serviceName := "test-service"
+	existingService := &ecs.Service{
+		TaskDefinition: aws.String("arn/test-task-def"),
+		Status:         aws.String("ACTIVE"),
+		DesiredCount:   aws.Int64(int64(existingDesiredCount)), // existing count > 1
+		ServiceName:    aws.String(serviceName),
+	}
+
+	// define expected client input given the above info
+	expectedInput := getDefaultUpdateInput()
+	expectedInput.serviceName = serviceName
+	expectedInput.count = *aws.Int64(int64(existingDesiredCount))
+
+	// call tests
+	upServiceWithCurrentTaskDefTest(t, cliContext, &config.CLIParams{}, &utils.ECSParams{}, expectedInput, existingService)
+	upServiceWithNewTaskDefTest(t, cliContext, &config.CLIParams{}, &utils.ECSParams{}, expectedInput, existingService)
+}
+
+func getDefaultUpdateInput() UpdateServiceParams {
+	return UpdateServiceParams{
+		deploymentConfig: &ecs.DeploymentConfiguration{}, //TODO: submit nil instead of empty obj on Update
+		count:            1,
+	}
+}
+
+func upServiceWithCurrentTaskDefTest(t *testing.T,
```

Comment:
I'm wondering if this and `upServiceWithNewTaskDefTest` could be DRY'd up a bit, since they are so similar?

Reply:
yeah, I think I can pull some of the task def set up and verification into separate functions (which can be reused by createServiceTest() as well). I don't want to mess around with much more since the intention is for "upServiceWithCurrentTaskDefTest" to be refactored out completely in the next couple PRs (after we remove 'UpdateServiceCount()'. Sound ok?

Reply:
Sounds super! (and yes in re: reuse by `createServiceTest()`) Thanks!